dashboard/src/components/settings/signInMethods/AzureADProviderSettings/AzureADProviderSettings.tsx

```diff
@@ -0,0 +1,203 @@
+    }),
+  clientSecret: Yup.string()
+    .label('Client Secret')
+    .when('enabled', {
+      is: true,
+      then: (schema) => schema.required(),
+    }),
+  tenant: Yup.string()
+    .label('Tenant')
+    .when('enabled', {
+      is: true,
+      then: (schema) => schema.required(),
+    }),
+  enabled: Yup.boolean(),
+});
+
+export type AzureADProviderFormValues = Yup.InferType<typeof validationSchema>;
+
+export default function AzureADProviderSettings() {
+  const { maintenanceActive } = useUI();
+  const { currentApplication } = useCurrentWorkspaceAndApplication();
+  const [updateConfig] = useUpdateConfigMutation({
+    refetchQueries: [GetSignInMethodsDocument],
+  });
+
+  const { data, loading, error } = useGetSignInMethodsQuery({
+    variables: { appId: currentApplication?.id },
+    fetchPolicy: 'cache-only',
+  });
+
+  const { clientId, clientSecret, tenant, enabled } =
+    data?.config?.auth?.method?.oauth?.azuread || {};
+
+  const form = useForm<AzureADProviderFormValues>({
+    reValidateMode: 'onSubmit',
+    defaultValues: {
+      clientId: clientId || '',
+      clientSecret: clientSecret || '',
+      tenant: tenant || '',
+      enabled: enabled || false,
+    },
+    resolver: yupResolver(validationSchema),
+  });
+
+  if (loading) {
+    return (
+      <ActivityIndicator
+        delay={1000}
+        label="Loading settings for AzureAD..."
+        className="justify-center"
+      />
+    );
+  }
+
+  if (error) {
+    throw error;
+  }
+
+  const { register, formState, watch } = form;
+  const authEnabled = watch('enabled');
+
+  const handleProviderUpdate = async (values: AzureADProviderFormValues) => {
+    const updateConfigPromise = updateConfig({
+      variables: {
+        appId: currentApplication.id,
+        config: {
+          auth: {
+            method: {
+              oauth: {
+                azuread: values,
+              },
+            },
+          },
+        },
+      },
+    });
+
+    try {
+      await toast.promise(
+        updateConfigPromise,
+        {
+          loading: `AzureAD settings are being updated...`,
+          success: `AzureAD settings have been updated successfully.`,
+          error: getServerError(
+            `An error occurred while trying to update the project's AzureAD settings.`,
+          ),
+        },
+        getToastStyleProps(),
+      );
+
+      form.reset(values);
+    } catch {
+      // Note: The toast will handle the error.
+    }
+  };
+
+  return (
+    <FormProvider {...form}>
+      <Form onSubmit={handleProviderUpdate}>
+        <SettingsContainer
+          title="AzureAD"
+          description="Allows users to sign in with AzureAD."
+          slotProps={{
+            submitButton: {
+              disabled: !formState.isDirty || maintenanceActive,
+              loading: formState.isSubmitting,
+            },
+          }}
+          docsLink="https://docs.nhost.io/authentication/sign-in-with-azuread"
```

Comment:
Could you please remove `docsLink` and `docsTitle`?  We'll add it later.

Reply:
Done!